conan_provider.cmake

```diff
@@ -35,6 +35,10 @@ set(CONAN_MINIMUM_VERSION 2.0.5)
 cmake_policy(PUSH)
 cmake_minimum_required(VERSION 3.24)
 
+# Handling of empty values for single-value keywords with cmake_parse_arguments()
+if(POLICY CMP0174)
+    cmake_policy(SET CMP0174 NEW)
+endif()
```

Comment:
Instead of this, we could update the `cmake_minimum_required()` call above to specify a version range that included up to at least 3.31. But that's a more involved change because it would require checking all policies added after CMake 3.24 up to whatever upper limit was specified. That's a separate focus to this PR, but would be good to do in a follow-up PR.